### PR TITLE
Use ride_id_t in more places

### PR DESCRIPTION
**Pull request by Gymnasiast**

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
Will need Replay rerecord and network bump

File: src/openrct2/rct12/RCT12.h [hunk 1]
```diff
@@ -14,6 +14,7 @@
 #include "../common.h"
 #include "../object/Object.h"
 
+#include <src/openrct2/ride/RideTypes.h>
```
Comment (duncanspumpkin) [inline src/openrct2/rct12/RCT12.h:17]: `../ride/RideTypes.h`

Comment (Gymnasiast) [reply src/openrct2/rct12/RCT12.h:17]: Argh, my IDE keeps auto-adding that :/

Comment (duncanspumpkin) [reply src/openrct2/rct12/RCT12.h:17]: I'll fix it never mind.

Comment (Gymnasiast) [reply src/openrct2/rct12/RCT12.h]: Sorry, read that too late - I already pushed a fix for that and for formatting.

Comment (duncanspumpkin) [reply src/openrct2/rct12/RCT12.h]: never mind you got there first.

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**